bindings/sundials4py/test/test_idas.py

```diff
@@ -0,0 +1,225 @@
+        return ode_problem.res(t, yy, yp, rr)
+
+    def psolve(t, yy, yp, rr, r, z, cj, delta, _):
+        return ode_problem.psolve(t, yy, yp, rr, r, z, cj, delta)
+
+    status = IDAInit(ida.get(), resfn, 0.0, yy, yp)
+    assert status == IDA_SUCCESS
+
+    status = IDASStolerances(ida.get(), 1e-4, 1e-4)
+    assert status == IDA_SUCCESS
+
+    status = IDASetLinearSolver(ida.get(), ls, None)
+    assert status == IDA_SUCCESS
+
+    status = IDASetPreconditioner(ida.get(), None, psolve)
+    assert status == IDA_SUCCESS
+
+    nrtfn = 2
+
+    def rootfn(t, yy, yp, gout, _):
+        # just a smoke test of the root finding callback
+        gout[:] = 1.0
+        assert len(gout) == nrtfn
+        return 0
+
+    status = IDARootInit(ida.get(), nrtfn, rootfn)
+    assert status == IDA_SUCCESS
+
+    tout = ode_problem.TF
+    status, tret = IDASolve(ida.get(), tout, yy, yp, IDA_NORMAL)
+    assert status == IDA_SUCCESS
+
+    status, num_steps = IDAGetNumSteps(ida.get())
+    assert status == IDA_SUCCESS
+    print("Number of steps: ", num_steps)
+
+    sol_yy = N_VClone(yy)
+    sol_yp = N_VClone(yp)
+
+    ode_problem.solution(sol_yy, sol_yp, tret)
+    assert np.allclose(N_VGetArrayPointer(sol_yy), N_VGetArrayPointer(yy), rtol=1e-2)
+    assert np.allclose(N_VGetArrayPointer(sol_yp), N_VGetArrayPointer(yp), rtol=1e-2)
+
+
+def test_idas_fsa(sunctx):
+    ode_problem = AnalyticDAE()
+
+    ida = IDACreate(sunctx)
+    yy = N_VNew_Serial(2, sunctx)
+    yp = N_VNew_Serial(2, sunctx)
+
+    # y and y' initial conditions
+    ode_problem.set_init_cond(yy, yp, ode_problem.T0)
+
+    ls = SUNLinSol_SPGMR(yy, SUN_PREC_LEFT, 0, sunctx)
+
+    def resfn(t, yy, yp, rr, _):
+        return ode_problem.res(t, yy, yp, rr)
+
+    def psolve(t, yy, yp, rr, r, z, cj, delta, _):
+        return ode_problem.psolve(t, yy, yp, rr, r, z, cj, delta)
+
+    status = IDAInit(ida.get(), resfn, 0.0, yy, yp)
+    assert status == IDA_SUCCESS
+
+    status = IDASStolerances(ida.get(), 1e-4, 1e-4)
+    assert status == IDA_SUCCESS
+
+    status = IDASetLinearSolver(ida.get(), ls, None)
+    assert status == IDA_SUCCESS
+
+    status = IDASetPreconditioner(ida.get(), None, psolve)
+    assert status == IDA_SUCCESS
+
+    # Sensitivity setup
+    Ns = 1  # number of sensitivities (wrt yy0)
+    ism = IDA_SIMULTANEOUS
+    yyS0 = [N_VClone(yy) for _ in range(Ns)]
+    ypS0 = [N_VClone(yp) for _ in range(Ns)]
+    for v in yyS0:
+        N_VConst(0.0, v)
+    for v in ypS0:
+        N_VConst(0.0, v)
+
+    def resS(Ns, t, yy, yp, resval, yS, ypS, resvalS, _, tmp1, tmp2, tmp3):
+        # Sensitivity residuals: d(res)/d(yy) * yyS + d(res)/d(yp) * ypS
+        # For smoke test, just zero out
+        for i in range(Ns):
+            N_VConst(0.0, resvalS[i])
+        return 0
+
+    status = IDASensInit(ida.get(), Ns, ism, resS, yyS0, ypS0)
+    assert status == IDA_SUCCESS
+
+    status = IDASensSStolerances(ida.get(), 1e-4, np.array([1e-4], dtype=sunrealtype))
```

Comment:
Use `SUNREALTYPE_RTOL` and `SUNREALTYPE_ATOL`?
```suggestion
    status = IDASensSStolerances(ida.get(), SUNREALTYPE_RTOL, np.array([SUNREALTYPE_ATOL], dtype=sunrealtype))
```